configure

```diff
@@ -2652,17 +2652,11 @@ case $_host_os in
 		exit 1
 		;;
 	darwin*)
-		# Pass -mlongcall to gcc so that it emits long calls
-		# which will allow for calls larger than 32MB. The linker
-		# will discard the calls if they are not needed, but we
-		# need to ensure the compiler emits them in the first place.
-		# Also the executable has grown to a size where using -Os is necessary to avoid a
-		# 'virtual memory exhausted' error when running the executable.
 		case $_host_cpu in
 		powerpc*)
-			append_var CFLAGS "-mlongcall"
```

Comment:
Why did you remove the `-mlongcall` option? Is it harmful?
It seems to help on AmigaOS4 where a dynamic modules build is done too.

Reply:
When switching to dynamic engine plugins on Mac PPC, the main (static) `scummvm` binary is now just 15MB (it was 52MB for ScummVM 2.2.0), so the `-mlongcall` flag for "calls larger than 32MB" appears unnecessary again, and I think we have enough room to avoid hitting this limit soon.

Adding it would mean that "all such calls take up one or two extra instructions" and I thought that it's better to keep the compiler/platform defaults whenever possible. Especially as the toolchain is a bit fragile there (the old Mach-O linker in particular is quite buggy/limited; and contrary to GNU binutils `ld`, nobody maintains the PPC Mach-O `ld` anymore; so the more you follow common usage the less you hit corner-case bugs).

As for AmigaOS4, I think it's maintained a bit differently, for example some toolchain flags appear to be used/kept "just in case" there. I think both approaches are fine for less-common platforms, it's just that the Mac PPC toolchain is not ELF and the Mach-O tools there are old, buggy and unmaintained, so I'd rather avoid non-default options if I don't need them and if they'd impact the Mach-O tools.

Plus, having the build fail at the very last linking stage after a 5-hour build because the `.o` files were built with something a bit more obscure that the Mach-O linker has troubles with is not fun ;) (and yeah I've lost some days to that in my previous attempts).